akka-actor/src/main/scala/akka/io/InetAddressDnsResolver.scala

```diff
@@ -94,22 +111,66 @@ class InetAddressDnsResolver(cache: SimpleDnsCache, config: Config) extends Acto
     }
   }
 
-  override def receive = {
+  override def receive: Receive = {
+    case DnsProtocol.Resolve(_, Srv) =>
+      sender() ! Status.Failure(
+        new IllegalArgumentException(
+          "SRV request sent to InetResolver. SRV requests are only supported by async-dns resolver"))
+
+    case r @ DnsProtocol.Resolve(name, ip @ Ip(ipv4, ipv6)) =>
+      val answer = cache.cached(r) match {
+        case Some(a) => a
+        case None =>
+          try {
+            val addresses: Array[InetAddress] = InetAddress.getAllByName(name)
+            val records = addressToRecords(name, addresses.toList, ipv4, ipv6)
+            val answer = DnsProtocol.Resolved(name, records.toList)
+            if (positiveCachePolicy != Never)
+              cache.put((name, Ip()), DnsProtocol.Resolved(name, records), positiveCachePolicy)
+            answer
+          } catch {
+            case _: UnknownHostException =>
+              val answer = DnsProtocol.Resolved(name, immutable.Seq.empty)
```

Comment:
What if records 1 and 2 resolve but 3 raises the exception, and we loose resolved, returning empty.

Reply:
What do you mean by records 1 and 2?